students/kristianjf/lesson09/mailroom_5.py

```diff
@@ -0,0 +1,221 @@
+                filenames.append(filename)
+        return filenames
+
+    def output_string(self):
+        '''Output string for letters'''
+        format_string = 'Dear {name},\n\nThank you for your generous donation of ' \
+                        '${donation}. Please send us more money at your earliest ' \
+                        'convenience.'
+        return format_string
+
+    #Function to Process Menu Options
+    def menu(self, options_dict):
+        '''
+        The script should prompt the user (you) to choose from a menu of 3 actions: \
+        “Send a Thank You”, “Create a Report” or “quit”)
+        '''
+        options = [option[0:2] for option in enumerate(options_dict.keys(), 1)]
+        while True:
+            print('Please select a number from the list of the following options: \n')
+            for option in options:
+                print(option)
+            response = self.menu_response_valid(options)
+            response_selection = options[response-1][1]
+            options_dict[response_selection]()
+
+    #Main Menu
+    def main_menu(self):
+        'Create Main Menu'
+        program_options_dict = {'Send a Thank You': self.thank_you_menu, \
+                                'Create a Report': self.create_a_report, \
+                                'Send Letters to Everyone': self.send_letters, 'quit': sys.exit}
+        self.menu(program_options_dict)
+
+    #Thank You Menu
+    def thank_you_menu(self):
+        '''
+        Sending a Thank You
+        If the user (you) selects ‘Send a Thank You’, prompt for a Full Name.
+        If the user types ‘list’, show them a list of the donor names and re-prompt
+        '''
+        program_options_ty = {'Send a Thank You': self.thank_you, \
+                            'Action: List': self.list_donors, \
+                            'quit': self.main_menu}
+        self.menu(program_options_ty)
+
+    #Menu Option Validation
+    def menu_response_valid(self, options):
+        'Validate Menu Options'
+        try:
+            response = int(input())
+            assert response in [option[0] for option in options]
+        except ValueError:
+            print('Non-integer value entered. Please try again.')
+        except AssertionError:
+            print('This number does not exist in the list of options. Please try again.')
+        else:
+            return response
+    #Name Validation
+    def name_response_valid(self, full_name):
+        'Validate and Return Name Input'
+        full_name_cap = ''
+        try:
+            for name in full_name:
+                assert name.isalpha()
+                full_name_cap += f'{name.capitalize()} '
+            full_name = full_name_cap.strip()
+        except AssertionError:
+            print('Invalid Name. Found non-alphabetic characters')
+        else:
+            return full_name
+
+def main():
+    'Create Donors'
+    jb = Donor('Jeff Bezos', [1, 5, 10])
```

Comment:
nice to have a function to create a populated DB, so it can be used in tests, too.